src/hotspot/os/linux/cgroupV1Subsystem_linux.hpp

```diff
@@ -51,50 +50,67 @@ class CgroupV1Controller: public CgroupController {
     char *subsystem_path() { return _path; }
 };
 
-class CgroupV1MemoryController: public CgroupV1Controller {
+class CgroupV1MemoryController : public CgroupMemoryController {
 
+  private:
+    CgroupV1Controller* _reader;
```

Comment:
It seems simpler to me to have this as a `CgroupV1Controller` and perhaps have a `reader()` accessor that returns a pointer. This class currently has a memory leak as the malloc-allocated `_reader` is never freed in the destructor. Up to you what you want to do.